### PR TITLE
Missing XmlDoco

### LINKED ISSUE #643: Remove <NoWarn>$(NoWarn);1591</NoWarn>
All public types should have XML doco and we should not disable this specific warning on a project level.

### PR DESCRIPTION
**Pull request by SeanFeldman**
Fixes #643 

@Particular/azure-maintainers please review

### CODE REVIEW

File: src/Transport/Addressing/Composition/Strategies/FlatComposition.cs [hunk 1]
```diff
@@ -2,11 +2,16 @@ namespace NServiceBus
 {
     using Transport.AzureServiceBus;
 
+    /// <summary>
+    /// Composition strategy resulting in a flat namespace hierarchy.
+    /// </summary>
     public class FlatComposition : ICompositionStrategy
     {
-        public string GetEntityPath(string entityname, EntityType entityType)
+        /// <summary></summary>
```
Comment (danielmarbach) [inline src/Transport/Addressing/Composition/Strategies/FlatComposition.cs]: Why is this empty?

File: src/Transport/Addressing/Individualization/IIndividualizationStrategy.cs [hunk 2]
```diff
@@ -1,7 +1,15 @@
 ﻿namespace NServiceBus.Transport.AzureServiceBus
 {
+    /// <summary>
+    /// Contract to implement a custom endpoint individualization.
+    /// </summary>
     public interface IIndividualizationStrategy
     {
+        /// <summary>
+        ///
```
Comment (danielmarbach) [inline src/Transport/Addressing/Individualization/IIndividualizationStrategy.cs]: Empty?

File: src/Transport/Addressing/Partitioning/Strategies/FailOverMode.cs [hunk 3]
```diff
@@ -1,8 +1,13 @@
 ﻿namespace NServiceBus.Transport.AzureServiceBus
 {
+    /// <summary>
+    /// Namespace in use for <see cref="FailOverNamespacePartitioning"/>.
+    /// </summary>
     public enum FailOverMode
     {
+        /// <summary></summary>
```
Comment (danielmarbach) [inline src/Transport/Addressing/Partitioning/Strategies/FailOverMode.cs]: Empty?

File: src/Transport/Addressing/Partitioning/Strategies/FailOverMode.cs [hunk 4]
```diff
@@ -1,8 +1,13 @@
...
         Primary,
+        /// <summary></summary>
```
Comment (danielmarbach) [inline src/Transport/Addressing/Partitioning/Strategies/FailOverMode.cs]: Empty

File: src/Transport/Addressing/Partitioning/Strategies/FailOverNamespacePartitioning.cs [hunk 5]
```diff
@@ -36,8 +40,12 @@ internal FailOverNamespacePartitioning(ReadOnlySettings settings)
             };
         }
 
+        /// <summary></summary>
```
Comment (danielmarbach) [inline src/Transport/Addressing/Partitioning/Strategies/FailOverNamespacePartitioning.cs]: Empty

File: src/Transport/Addressing/Sanitization/Strategies/ThrowOnFailedValidation.cs [hunk 6]
```diff
@@ -103,6 +107,7 @@ internal ThrowOnFailedValidation(ReadOnlySettings settings)
             }
         }
 
+        /// <summary></summary>
```
Comment (danielmarbach) [inline src/Transport/Addressing/Sanitization/Strategies/ThrowOnFailedValidation.cs]: empty

File: src/Transport/Addressing/Sanitization/Strategies/ValidateAndHashIfNeeded.cs [hunk 7]
```diff
@@ -125,6 +129,7 @@ internal ValidateAndHashIfNeeded(ReadOnlySettings settings)
             }
         }
 
+        /// <summary></summary>
```
Comment (danielmarbach) [inline src/Transport/Addressing/Sanitization/Strategies/ValidateAndHashIfNeeded.cs]: empty

File: src/Transport/Addressing/Sanitization/ValidationResult.cs [hunk 8]
```diff
@@ -1,29 +1,40 @@
 ﻿namespace NServiceBus.Transport.AzureServiceBus
 {
+    /// <summary>
+    /// Validation result to provide information about validation performed by the configured <see cref="ISanitizationStrategy"/>.
+    /// </summary>
     public class ValidationResult
     {
+        /// <summary></summary>
```
Comment (danielmarbach) [inline src/Transport/Addressing/Sanitization/ValidationResult.cs]: All empty comments?

File: src/Transport/Config/ExtensionPoints/AzureServiceBusCompositionExtensionPoint.cs [hunk 9]
```diff
@@ -4,6 +4,7 @@
     using Settings;
     using Transport.AzureServiceBus;
 
+    /// <summary></summary>
```
Comment (danielmarbach) [inline src/Transport/Config/ExtensionPoints/AzureServiceBusCompositionExtensionPoint.cs]: empty

File: src/Transport/Config/ExtensionPoints/AzureServiceBusDiscriminatorBasedIndividualizationSettingsExtensions.cs [hunk 10]
```diff
@@ -4,6 +4,7 @@
     using Configuration.AdvancedExtensibility;
     using Transport.AzureServiceBus;
 
+    /// <summary></summary>
```
Comment (danielmarbach) [inline src/Transport/Config/ExtensionPoints/AzureServiceBusDiscriminatorBasedIndividualizationSettingsExtensions.cs]: empty

File: src/Transport/Config/ExtensionPoints/AzureServiceBusEndpointOrientedTopologySettingsExtensions.cs [hunk 11]
```diff
@@ -7,14 +7,21 @@
     using Settings;
     using Transport.AzureServiceBus;
 
+    /// <summary></summary>
     public static partial class AzureServiceBusEndpointOrientedTopologySettingsExtensions
     {
+        /// <summary>Register publisher endpoint name for a give event type.</summary>
+        /// <param name="topologySettings"></param>
```
Comment (danielmarbach) [inline src/Transport/Config/ExtensionPoints/AzureServiceBusEndpointOrientedTopologySettingsExtensions.cs]: why not just remove the params here when not documented?

File: src/Transport/Config/ExtensionPoints/AzureServiceBusEndpointOrientedTopologySettingsExtensions.cs [hunk 12]
```diff
@@ -7,14 +7,21 @@
     using Settings;
     using Transport.AzureServiceBus;
 
+    /// <summary></summary>
```
Comment (danielmarbach) [inline src/Transport/Config/ExtensionPoints/AzureServiceBusEndpointOrientedTopologySettingsExtensions.cs]: Empty

File: src/Transport/Config/ExtensionPoints/AzureServiceBusIndividualizationExtensionPoint.cs [hunk 13]
```diff
@@ -4,6 +4,7 @@
     using Settings;
     using Transport.AzureServiceBus;
 
+    /// <summary></summary>
```
Comment (danielmarbach) [inline src/Transport/Config/ExtensionPoints/AzureServiceBusIndividualizationExtensionPoint.cs]: empty

File: src/Transport/Receiving/UnsupportedBrokeredMessageBodyTypeException.cs [hunk 14]
```diff
@@ -2,22 +2,30 @@
...
     using System.Runtime.Serialization;
+    using Microsoft.ServiceBus.Messaging;
 
+    /// <summary>
+    /// Exception thrown when an unsupported <see cref="BrokeredMessage"/> encoding is encountered.
+    /// </summary>
     [Serializable]
     public class UnsupportedBrokeredMessageBodyTypeException : Exception
     {
+        /// <summary></summary>
```
Comment (danielmarbach) [inline src/Transport/Receiving/UnsupportedBrokeredMessageBodyTypeException.cs]: empty?

File: src/Transport/Sending/MessageTooLargeException.cs [hunk 15]
```diff
@@ -2,22 +2,30 @@ namespace NServiceBus.Transport.AzureServiceBus
...
     using System.Runtime.Serialization;
+    using Microsoft.ServiceBus.Messaging;
 
+    /// <summary>
+    /// <see cref="BrokeredMessage"/> is too large to be sent.
+    /// </summary>
     [Serializable]
     public class MessageTooLargeException : Exception
     {
+        /// <summary> </summary>
```
Comment (danielmarbach) [inline src/Transport/Sending/MessageTooLargeException.cs]: empty?

File: src/Transport/Topology/MetaModel/EntityType.cs [hunk 16]
```diff
@@ -1,10 +1,15 @@
 namespace NServiceBus
 {
+    /// <summary></summary>
```
Comment (danielmarbach) [inline src/Transport/Topology/MetaModel/EntityType.cs]: All empty?

File: src/Transport/Topology/MetaModel/NamespaceConfigurations.cs [hunk 17]
```diff
@@ -6,8 +6,10 @@
     using System.Linq;
     using Logging;
 
+    /// <summary> </summary>
```
Comment (danielmarbach) [inline src/Transport/Topology/MetaModel/NamespaceConfigurations.cs]: All empty?

File: src/Transport/Config/ExtensionPoints/AzureServiceBusEndpointOrientedTopologySettingsExtensions.cs (hunk 12 above)
Comment (SeanFeldman) [reply src/Transport/Config/ExtensionPoints/AzureServiceBusEndpointOrientedTopologySettingsExtensions.cs]: Missed that one. Thanks.

File: src/Transport/Config/ExtensionPoints/AzureServiceBusDiscriminatorBasedIndividualizationSettingsExtensions.cs (hunk 10 above)
Comment (SeanFeldman) [reply src/Transport/Config/ExtensionPoints/AzureServiceBusDiscriminatorBasedIndividualizationSettingsExtensions.cs]: Missed that one. Thanks.

File: src/Transport/Addressing/Sanitization/ValidationResult.cs (hunk 8 above)
Comment (SeanFeldman) [reply src/Transport/Addressing/Sanitization/ValidationResult.cs]: This should never be exposed to begin with, but it is 🙁

File: src/Transport/Addressing/Sanitization/Strategies/ValidateAndHashIfNeeded.cs (hunk 7 above)
Comment (SeanFeldman) [reply src/Transport/Addressing/Sanitization/Strategies/ValidateAndHashIfNeeded.cs]: Find it redundant to explain the obvious, but will change.

File: src/Transport/Addressing/Sanitization/Strategies/ThrowOnFailedValidation.cs (hunk 6 above)
Comment (SeanFeldman) [reply src/Transport/Addressing/Sanitization/Strategies/ThrowOnFailedValidation.cs]: In this case fully agree, it should be documented. `Sanitize()` here has a hidden behavior that is not obvious. Done.

File: src/Transport/Addressing/Partitioning/Strategies/FailOverNamespacePartitioning.cs (hunk 5 above)
Comment (SeanFeldman) [reply src/Transport/Addressing/Partitioning/Strategies/FailOverNamespacePartitioning.cs]: This should never be exposed...

**Comment by SeanFeldman** [comment]
@danielmarbach - addressed.

Considering that we went a full version w/o xmldoco, let's not go crazy about xmldoco 😃 

Also, there are some that I consider to be exceptional. For example:

`MessageTooLargeException` constructor - given that the type is already defining what this exception is
```
/// <summary>
/// <see cref="BrokeredMessage"/> is too large to be sent.
/// </summary>
[Serializable]
public class MessageTooLargeException : Exception
```
what else would you expect to see? I don't even think we needed a specialized class as there's nothing a used can do about it when this exception is thrown.

(`UnsupportedBrokeredMessageBodyTypeException` is the same).

Or documenting the obvious such as `EntityType`

```c#
    /// <summary>Entity type.</summary>
    public enum EntityType
    {
        /// <summary>Queue</summary>
        Queue,
        /// <summary>Topic</summary>
        Topic,
        /// <summary>Subscription</summary>
        Subscription,
        /// <summary>Rule</summary>
        Rule
    }
```

**Pull request merged**